diff -rupN gcc-4.9.2/gcc/config/avr/avr-mcus.def gcc-4.9.2-patched/gcc/config/avr/avr-mcus.def
--- gcc-4.9.2/gcc/config/avr/avr-mcus.def	2014-05-09 13:52:20.000000000 +0200
+++ gcc-4.9.2-patched/gcc/config/avr/avr-mcus.def	2014-12-18 19:28:53.000000000 +0100
@@ -83,8 +83,10 @@ AVR_MCU ("attiny24a",            ARCH_AV
 AVR_MCU ("attiny4313",           ARCH_AVR25, AVR_ISA_NONE, "__AVR_ATtiny4313__",       0x0060, 1, "tn4313")
 AVR_MCU ("attiny44",             ARCH_AVR25, AVR_ISA_NONE, "__AVR_ATtiny44__",         0x0060, 1, "tn44")
 AVR_MCU ("attiny44a",            ARCH_AVR25, AVR_ISA_NONE, "__AVR_ATtiny44A__",        0x0060, 1, "tn44a")
+AVR_MCU ("attiny441",            ARCH_AVR25, AVR_ISA_NONE, "__AVR_ATtiny441__",        0x0100, 1, "tn441")
 AVR_MCU ("attiny84",             ARCH_AVR25, AVR_ISA_NONE, "__AVR_ATtiny84__",         0x0060, 1, "tn84")
 AVR_MCU ("attiny84a",            ARCH_AVR25, AVR_ISA_NONE, "__AVR_ATtiny84A__",        0x0060, 1, "tn84")
+AVR_MCU ("attiny841",            ARCH_AVR25, AVR_ISA_NONE, "__AVR_ATtiny841__",        0x0100, 1, "tn841")
 AVR_MCU ("attiny25",             ARCH_AVR25, AVR_SHORT_SP, "__AVR_ATtiny25__",         0x0060, 1, "tn25")
 AVR_MCU ("attiny45",             ARCH_AVR25, AVR_ISA_NONE, "__AVR_ATtiny45__",         0x0060, 1, "tn45")
 AVR_MCU ("attiny85",             ARCH_AVR25, AVR_ISA_NONE, "__AVR_ATtiny85__",         0x0060, 1, "tn85")
diff -rupN gcc-4.9.2/gcc/config/avr/avr-tables.opt gcc-4.9.2-patched/gcc/config/avr/avr-tables.opt
--- gcc-4.9.2/gcc/config/avr/avr-tables.opt	2014-10-30 09:28:58.000000000 +0100
+++ gcc-4.9.2-patched/gcc/config/avr/avr-tables.opt	2014-12-18 19:29:22.000000000 +0100
@@ -99,668 +99,674 @@ EnumValue
 Enum(avr_mcu) String(attiny44a) Value(24)
 
 EnumValue
-Enum(avr_mcu) String(attiny84) Value(25)
+Enum(avr_mcu) String(attiny441) Value(25)
 
 EnumValue
-Enum(avr_mcu) String(attiny84a) Value(26)
+Enum(avr_mcu) String(attiny84) Value(26)
 
 EnumValue
-Enum(avr_mcu) String(attiny25) Value(27)
+Enum(avr_mcu) String(attiny84a) Value(27)
 
 EnumValue
-Enum(avr_mcu) String(attiny45) Value(28)
+Enum(avr_mcu) String(attiny841) Value(28)
 
 EnumValue
-Enum(avr_mcu) String(attiny85) Value(29)
+Enum(avr_mcu) String(attiny25) Value(29)
 
 EnumValue
-Enum(avr_mcu) String(attiny261) Value(30)
+Enum(avr_mcu) String(attiny45) Value(30)
 
 EnumValue
-Enum(avr_mcu) String(attiny261a) Value(31)
+Enum(avr_mcu) String(attiny85) Value(31)
 
 EnumValue
-Enum(avr_mcu) String(attiny461) Value(32)
+Enum(avr_mcu) String(attiny261) Value(32)
 
 EnumValue
-Enum(avr_mcu) String(attiny461a) Value(33)
+Enum(avr_mcu) String(attiny261a) Value(33)
 
 EnumValue
-Enum(avr_mcu) String(attiny861) Value(34)
+Enum(avr_mcu) String(attiny461) Value(34)
 
 EnumValue
-Enum(avr_mcu) String(attiny861a) Value(35)
+Enum(avr_mcu) String(attiny461a) Value(35)
 
 EnumValue
-Enum(avr_mcu) String(attiny43u) Value(36)
+Enum(avr_mcu) String(attiny861) Value(36)
 
 EnumValue
-Enum(avr_mcu) String(attiny87) Value(37)
+Enum(avr_mcu) String(attiny861a) Value(37)
 
 EnumValue
-Enum(avr_mcu) String(attiny48) Value(38)
+Enum(avr_mcu) String(attiny43u) Value(38)
 
 EnumValue
-Enum(avr_mcu) String(attiny88) Value(39)
+Enum(avr_mcu) String(attiny87) Value(39)
 
 EnumValue
-Enum(avr_mcu) String(at86rf401) Value(40)
+Enum(avr_mcu) String(attiny48) Value(40)
 
 EnumValue
-Enum(avr_mcu) String(avr3) Value(41)
+Enum(avr_mcu) String(attiny88) Value(41)
 
 EnumValue
-Enum(avr_mcu) String(at43usb355) Value(42)
+Enum(avr_mcu) String(at86rf401) Value(42)
 
 EnumValue
-Enum(avr_mcu) String(at76c711) Value(43)
+Enum(avr_mcu) String(avr3) Value(43)
 
 EnumValue
-Enum(avr_mcu) String(avr31) Value(44)
+Enum(avr_mcu) String(at43usb355) Value(44)
 
 EnumValue
-Enum(avr_mcu) String(atmega103) Value(45)
+Enum(avr_mcu) String(at76c711) Value(45)
 
 EnumValue
-Enum(avr_mcu) String(at43usb320) Value(46)
+Enum(avr_mcu) String(avr31) Value(46)
 
 EnumValue
-Enum(avr_mcu) String(avr35) Value(47)
+Enum(avr_mcu) String(atmega103) Value(47)
 
 EnumValue
-Enum(avr_mcu) String(ata5505) Value(48)
+Enum(avr_mcu) String(at43usb320) Value(48)
 
 EnumValue
-Enum(avr_mcu) String(at90usb82) Value(49)
+Enum(avr_mcu) String(avr35) Value(49)
 
 EnumValue
-Enum(avr_mcu) String(at90usb162) Value(50)
+Enum(avr_mcu) String(ata5505) Value(50)
 
 EnumValue
-Enum(avr_mcu) String(atmega8u2) Value(51)
+Enum(avr_mcu) String(at90usb82) Value(51)
 
 EnumValue
-Enum(avr_mcu) String(atmega16u2) Value(52)
+Enum(avr_mcu) String(at90usb162) Value(52)
 
 EnumValue
-Enum(avr_mcu) String(atmega32u2) Value(53)
+Enum(avr_mcu) String(atmega8u2) Value(53)
 
 EnumValue
-Enum(avr_mcu) String(attiny167) Value(54)
+Enum(avr_mcu) String(atmega16u2) Value(54)
 
 EnumValue
-Enum(avr_mcu) String(attiny1634) Value(55)
+Enum(avr_mcu) String(atmega32u2) Value(55)
 
 EnumValue
-Enum(avr_mcu) String(avr4) Value(56)
+Enum(avr_mcu) String(attiny167) Value(56)
 
 EnumValue
-Enum(avr_mcu) String(ata6285) Value(57)
+Enum(avr_mcu) String(attiny1634) Value(57)
 
 EnumValue
-Enum(avr_mcu) String(ata6286) Value(58)
+Enum(avr_mcu) String(avr4) Value(58)
 
 EnumValue
-Enum(avr_mcu) String(atmega8) Value(59)
+Enum(avr_mcu) String(ata6285) Value(59)
 
 EnumValue
-Enum(avr_mcu) String(atmega8a) Value(60)
+Enum(avr_mcu) String(ata6286) Value(60)
 
 EnumValue
-Enum(avr_mcu) String(atmega48) Value(61)
+Enum(avr_mcu) String(atmega8) Value(61)
 
 EnumValue
-Enum(avr_mcu) String(atmega48a) Value(62)
+Enum(avr_mcu) String(atmega8a) Value(62)
 
 EnumValue
-Enum(avr_mcu) String(atmega48p) Value(63)
+Enum(avr_mcu) String(atmega48) Value(63)
 
 EnumValue
-Enum(avr_mcu) String(atmega48pa) Value(64)
+Enum(avr_mcu) String(atmega48a) Value(64)
 
 EnumValue
-Enum(avr_mcu) String(atmega88) Value(65)
+Enum(avr_mcu) String(atmega48p) Value(65)
 
 EnumValue
-Enum(avr_mcu) String(atmega88a) Value(66)
+Enum(avr_mcu) String(atmega48pa) Value(66)
 
 EnumValue
-Enum(avr_mcu) String(atmega88p) Value(67)
+Enum(avr_mcu) String(atmega88) Value(67)
 
 EnumValue
-Enum(avr_mcu) String(atmega88pa) Value(68)
+Enum(avr_mcu) String(atmega88a) Value(68)
 
 EnumValue
-Enum(avr_mcu) String(atmega8515) Value(69)
+Enum(avr_mcu) String(atmega88p) Value(69)
 
 EnumValue
-Enum(avr_mcu) String(atmega8535) Value(70)
+Enum(avr_mcu) String(atmega88pa) Value(70)
 
 EnumValue
-Enum(avr_mcu) String(atmega8hva) Value(71)
+Enum(avr_mcu) String(atmega8515) Value(71)
 
 EnumValue
-Enum(avr_mcu) String(at90pwm1) Value(72)
+Enum(avr_mcu) String(atmega8535) Value(72)
 
 EnumValue
-Enum(avr_mcu) String(at90pwm2) Value(73)
+Enum(avr_mcu) String(atmega8hva) Value(73)
 
 EnumValue
-Enum(avr_mcu) String(at90pwm2b) Value(74)
+Enum(avr_mcu) String(at90pwm1) Value(74)
 
 EnumValue
-Enum(avr_mcu) String(at90pwm3) Value(75)
+Enum(avr_mcu) String(at90pwm2) Value(75)
 
 EnumValue
-Enum(avr_mcu) String(at90pwm3b) Value(76)
+Enum(avr_mcu) String(at90pwm2b) Value(76)
 
 EnumValue
-Enum(avr_mcu) String(at90pwm81) Value(77)
+Enum(avr_mcu) String(at90pwm3) Value(77)
 
 EnumValue
-Enum(avr_mcu) String(avr5) Value(78)
+Enum(avr_mcu) String(at90pwm3b) Value(78)
 
 EnumValue
-Enum(avr_mcu) String(ata5790) Value(79)
+Enum(avr_mcu) String(at90pwm81) Value(79)
 
 EnumValue
-Enum(avr_mcu) String(ata5790n) Value(80)
+Enum(avr_mcu) String(avr5) Value(80)
 
 EnumValue
-Enum(avr_mcu) String(ata5795) Value(81)
+Enum(avr_mcu) String(ata5790) Value(81)
 
 EnumValue
-Enum(avr_mcu) String(atmega16) Value(82)
+Enum(avr_mcu) String(ata5790n) Value(82)
 
 EnumValue
-Enum(avr_mcu) String(atmega16a) Value(83)
+Enum(avr_mcu) String(ata5795) Value(83)
 
 EnumValue
-Enum(avr_mcu) String(atmega161) Value(84)
+Enum(avr_mcu) String(atmega16) Value(84)
 
 EnumValue
-Enum(avr_mcu) String(atmega162) Value(85)
+Enum(avr_mcu) String(atmega16a) Value(85)
 
 EnumValue
-Enum(avr_mcu) String(atmega163) Value(86)
+Enum(avr_mcu) String(atmega161) Value(86)
 
 EnumValue
-Enum(avr_mcu) String(atmega164a) Value(87)
+Enum(avr_mcu) String(atmega162) Value(87)
 
 EnumValue
-Enum(avr_mcu) String(atmega164p) Value(88)
+Enum(avr_mcu) String(atmega163) Value(88)
 
 EnumValue
-Enum(avr_mcu) String(atmega164pa) Value(89)
+Enum(avr_mcu) String(atmega164a) Value(89)
 
 EnumValue
-Enum(avr_mcu) String(atmega165) Value(90)
+Enum(avr_mcu) String(atmega164p) Value(90)
 
 EnumValue
-Enum(avr_mcu) String(atmega165a) Value(91)
+Enum(avr_mcu) String(atmega164pa) Value(91)
 
 EnumValue
-Enum(avr_mcu) String(atmega165p) Value(92)
+Enum(avr_mcu) String(atmega165) Value(92)
 
 EnumValue
-Enum(avr_mcu) String(atmega165pa) Value(93)
+Enum(avr_mcu) String(atmega165a) Value(93)
 
 EnumValue
-Enum(avr_mcu) String(atmega168) Value(94)
+Enum(avr_mcu) String(atmega165p) Value(94)
 
 EnumValue
-Enum(avr_mcu) String(atmega168a) Value(95)
+Enum(avr_mcu) String(atmega165pa) Value(95)
 
 EnumValue
-Enum(avr_mcu) String(atmega168p) Value(96)
+Enum(avr_mcu) String(atmega168) Value(96)
 
 EnumValue
-Enum(avr_mcu) String(atmega168pa) Value(97)
+Enum(avr_mcu) String(atmega168a) Value(97)
 
 EnumValue
-Enum(avr_mcu) String(atmega169) Value(98)
+Enum(avr_mcu) String(atmega168p) Value(98)
 
 EnumValue
-Enum(avr_mcu) String(atmega169a) Value(99)
+Enum(avr_mcu) String(atmega168pa) Value(99)
 
 EnumValue
-Enum(avr_mcu) String(atmega169p) Value(100)
+Enum(avr_mcu) String(atmega169) Value(100)
 
 EnumValue
-Enum(avr_mcu) String(atmega169pa) Value(101)
+Enum(avr_mcu) String(atmega169a) Value(101)
 
 EnumValue
-Enum(avr_mcu) String(atmega16hvb) Value(102)
+Enum(avr_mcu) String(atmega169p) Value(102)
 
 EnumValue
-Enum(avr_mcu) String(atmega16hvbrevb) Value(103)
+Enum(avr_mcu) String(atmega169pa) Value(103)
 
 EnumValue
-Enum(avr_mcu) String(atmega16m1) Value(104)
+Enum(avr_mcu) String(atmega16hvb) Value(104)
 
 EnumValue
-Enum(avr_mcu) String(atmega16u4) Value(105)
+Enum(avr_mcu) String(atmega16hvbrevb) Value(105)
 
 EnumValue
-Enum(avr_mcu) String(atmega26hvg) Value(106)
+Enum(avr_mcu) String(atmega16m1) Value(106)
 
 EnumValue
-Enum(avr_mcu) String(atmega32a) Value(107)
+Enum(avr_mcu) String(atmega16u4) Value(107)
 
 EnumValue
-Enum(avr_mcu) String(atmega32) Value(108)
+Enum(avr_mcu) String(atmega26hvg) Value(108)
 
 EnumValue
-Enum(avr_mcu) String(atmega323) Value(109)
+Enum(avr_mcu) String(atmega32a) Value(109)
 
 EnumValue
-Enum(avr_mcu) String(atmega324a) Value(110)
+Enum(avr_mcu) String(atmega32) Value(110)
 
 EnumValue
-Enum(avr_mcu) String(atmega324p) Value(111)
+Enum(avr_mcu) String(atmega323) Value(111)
 
 EnumValue
-Enum(avr_mcu) String(atmega324pa) Value(112)
+Enum(avr_mcu) String(atmega324a) Value(112)
 
 EnumValue
-Enum(avr_mcu) String(atmega325) Value(113)
+Enum(avr_mcu) String(atmega324p) Value(113)
 
 EnumValue
-Enum(avr_mcu) String(atmega325a) Value(114)
+Enum(avr_mcu) String(atmega324pa) Value(114)
 
 EnumValue
-Enum(avr_mcu) String(atmega325p) Value(115)
+Enum(avr_mcu) String(atmega325) Value(115)
 
 EnumValue
-Enum(avr_mcu) String(atmega3250) Value(116)
+Enum(avr_mcu) String(atmega325a) Value(116)
 
 EnumValue
-Enum(avr_mcu) String(atmega3250a) Value(117)
+Enum(avr_mcu) String(atmega325p) Value(117)
 
 EnumValue
-Enum(avr_mcu) String(atmega3250p) Value(118)
+Enum(avr_mcu) String(atmega3250) Value(118)
 
 EnumValue
-Enum(avr_mcu) String(atmega3250pa) Value(119)
+Enum(avr_mcu) String(atmega3250a) Value(119)
 
 EnumValue
-Enum(avr_mcu) String(atmega328) Value(120)
+Enum(avr_mcu) String(atmega3250p) Value(120)
 
 EnumValue
-Enum(avr_mcu) String(atmega328p) Value(121)
+Enum(avr_mcu) String(atmega3250pa) Value(121)
 
 EnumValue
-Enum(avr_mcu) String(atmega329) Value(122)
+Enum(avr_mcu) String(atmega328) Value(122)
 
 EnumValue
-Enum(avr_mcu) String(atmega329a) Value(123)
+Enum(avr_mcu) String(atmega328p) Value(123)
 
 EnumValue
-Enum(avr_mcu) String(atmega329p) Value(124)
+Enum(avr_mcu) String(atmega329) Value(124)
 
 EnumValue
-Enum(avr_mcu) String(atmega329pa) Value(125)
+Enum(avr_mcu) String(atmega329a) Value(125)
 
 EnumValue
-Enum(avr_mcu) String(atmega3290) Value(126)
+Enum(avr_mcu) String(atmega329p) Value(126)
 
 EnumValue
-Enum(avr_mcu) String(atmega3290a) Value(127)
+Enum(avr_mcu) String(atmega329pa) Value(127)
 
 EnumValue
-Enum(avr_mcu) String(atmega3290p) Value(128)
+Enum(avr_mcu) String(atmega3290) Value(128)
 
 EnumValue
-Enum(avr_mcu) String(atmega3290pa) Value(129)
+Enum(avr_mcu) String(atmega3290a) Value(129)
 
 EnumValue
-Enum(avr_mcu) String(atmega32c1) Value(130)
+Enum(avr_mcu) String(atmega3290p) Value(130)
 
 EnumValue
-Enum(avr_mcu) String(atmega32m1) Value(131)
+Enum(avr_mcu) String(atmega3290pa) Value(131)
 
 EnumValue
-Enum(avr_mcu) String(atmega32u4) Value(132)
+Enum(avr_mcu) String(atmega32c1) Value(132)
 
 EnumValue
-Enum(avr_mcu) String(atmega32u6) Value(133)
+Enum(avr_mcu) String(atmega32m1) Value(133)
 
 EnumValue
-Enum(avr_mcu) String(atmega406) Value(134)
+Enum(avr_mcu) String(atmega32u4) Value(134)
 
 EnumValue
-Enum(avr_mcu) String(atmega64) Value(135)
+Enum(avr_mcu) String(atmega32u6) Value(135)
 
 EnumValue
-Enum(avr_mcu) String(atmega64a) Value(136)
+Enum(avr_mcu) String(atmega406) Value(136)
 
 EnumValue
-Enum(avr_mcu) String(atmega640) Value(137)
+Enum(avr_mcu) String(atmega64) Value(137)
 
 EnumValue
-Enum(avr_mcu) String(atmega644) Value(138)
+Enum(avr_mcu) String(atmega64a) Value(138)
 
 EnumValue
-Enum(avr_mcu) String(atmega644a) Value(139)
+Enum(avr_mcu) String(atmega640) Value(139)
 
 EnumValue
-Enum(avr_mcu) String(atmega644p) Value(140)
+Enum(avr_mcu) String(atmega644) Value(140)
 
 EnumValue
-Enum(avr_mcu) String(atmega644pa) Value(141)
+Enum(avr_mcu) String(atmega644a) Value(141)
 
 EnumValue
-Enum(avr_mcu) String(atmega645) Value(142)
+Enum(avr_mcu) String(atmega644p) Value(142)
 
 EnumValue
-Enum(avr_mcu) String(atmega645a) Value(143)
+Enum(avr_mcu) String(atmega644pa) Value(143)
 
 EnumValue
-Enum(avr_mcu) String(atmega645p) Value(144)
+Enum(avr_mcu) String(atmega645) Value(144)
 
 EnumValue
-Enum(avr_mcu) String(atmega6450) Value(145)
+Enum(avr_mcu) String(atmega645a) Value(145)
 
 EnumValue
-Enum(avr_mcu) String(atmega6450a) Value(146)
+Enum(avr_mcu) String(atmega645p) Value(146)
 
 EnumValue
-Enum(avr_mcu) String(atmega6450p) Value(147)
+Enum(avr_mcu) String(atmega6450) Value(147)
 
 EnumValue
-Enum(avr_mcu) String(atmega649) Value(148)
+Enum(avr_mcu) String(atmega6450a) Value(148)
 
 EnumValue
-Enum(avr_mcu) String(atmega649a) Value(149)
+Enum(avr_mcu) String(atmega6450p) Value(149)
 
 EnumValue
-Enum(avr_mcu) String(atmega649p) Value(150)
+Enum(avr_mcu) String(atmega649) Value(150)
 
 EnumValue
-Enum(avr_mcu) String(atmega6490) Value(151)
+Enum(avr_mcu) String(atmega649a) Value(151)
 
 EnumValue
-Enum(avr_mcu) String(atmega16hva) Value(152)
+Enum(avr_mcu) String(atmega649p) Value(152)
 
 EnumValue
-Enum(avr_mcu) String(atmega16hva2) Value(153)
+Enum(avr_mcu) String(atmega6490) Value(153)
 
 EnumValue
-Enum(avr_mcu) String(atmega32hvb) Value(154)
+Enum(avr_mcu) String(atmega16hva) Value(154)
 
 EnumValue
-Enum(avr_mcu) String(atmega6490a) Value(155)
+Enum(avr_mcu) String(atmega16hva2) Value(155)
 
 EnumValue
-Enum(avr_mcu) String(atmega6490p) Value(156)
+Enum(avr_mcu) String(atmega32hvb) Value(156)
 
 EnumValue
-Enum(avr_mcu) String(atmega64c1) Value(157)
+Enum(avr_mcu) String(atmega6490a) Value(157)
 
 EnumValue
-Enum(avr_mcu) String(atmega64m1) Value(158)
+Enum(avr_mcu) String(atmega6490p) Value(158)
 
 EnumValue
-Enum(avr_mcu) String(atmega64hve) Value(159)
+Enum(avr_mcu) String(atmega64c1) Value(159)
 
 EnumValue
-Enum(avr_mcu) String(atmega64rfa2) Value(160)
+Enum(avr_mcu) String(atmega64m1) Value(160)
 
 EnumValue
-Enum(avr_mcu) String(atmega64rfr2) Value(161)
+Enum(avr_mcu) String(atmega64hve) Value(161)
 
 EnumValue
-Enum(avr_mcu) String(atmega32hvbrevb) Value(162)
+Enum(avr_mcu) String(atmega64rfa2) Value(162)
 
 EnumValue
-Enum(avr_mcu) String(atmega48hvf) Value(163)
+Enum(avr_mcu) String(atmega64rfr2) Value(163)
 
 EnumValue
-Enum(avr_mcu) String(at90can32) Value(164)
+Enum(avr_mcu) String(atmega32hvbrevb) Value(164)
 
 EnumValue
-Enum(avr_mcu) String(at90can64) Value(165)
+Enum(avr_mcu) String(atmega48hvf) Value(165)
 
 EnumValue
-Enum(avr_mcu) String(at90pwm161) Value(166)
+Enum(avr_mcu) String(at90can32) Value(166)
 
 EnumValue
-Enum(avr_mcu) String(at90pwm216) Value(167)
+Enum(avr_mcu) String(at90can64) Value(167)
 
 EnumValue
-Enum(avr_mcu) String(at90pwm316) Value(168)
+Enum(avr_mcu) String(at90pwm161) Value(168)
 
 EnumValue
-Enum(avr_mcu) String(at90scr100) Value(169)
+Enum(avr_mcu) String(at90pwm216) Value(169)
 
 EnumValue
-Enum(avr_mcu) String(at90usb646) Value(170)
+Enum(avr_mcu) String(at90pwm316) Value(170)
 
 EnumValue
-Enum(avr_mcu) String(at90usb647) Value(171)
+Enum(avr_mcu) String(at90scr100) Value(171)
 
 EnumValue
-Enum(avr_mcu) String(at94k) Value(172)
+Enum(avr_mcu) String(at90usb646) Value(172)
 
 EnumValue
-Enum(avr_mcu) String(m3000) Value(173)
+Enum(avr_mcu) String(at90usb647) Value(173)
 
 EnumValue
-Enum(avr_mcu) String(avr51) Value(174)
+Enum(avr_mcu) String(at94k) Value(174)
 
 EnumValue
-Enum(avr_mcu) String(atmega128) Value(175)
+Enum(avr_mcu) String(m3000) Value(175)
 
 EnumValue
-Enum(avr_mcu) String(atmega128a) Value(176)
+Enum(avr_mcu) String(avr51) Value(176)
 
 EnumValue
-Enum(avr_mcu) String(atmega1280) Value(177)
+Enum(avr_mcu) String(atmega128) Value(177)
 
 EnumValue
-Enum(avr_mcu) String(atmega1281) Value(178)
+Enum(avr_mcu) String(atmega128a) Value(178)
 
 EnumValue
-Enum(avr_mcu) String(atmega1284) Value(179)
+Enum(avr_mcu) String(atmega1280) Value(179)
 
 EnumValue
-Enum(avr_mcu) String(atmega1284p) Value(180)
+Enum(avr_mcu) String(atmega1281) Value(180)
 
 EnumValue
-Enum(avr_mcu) String(atmega128rfa1) Value(181)
+Enum(avr_mcu) String(atmega1284) Value(181)
 
 EnumValue
-Enum(avr_mcu) String(at90can128) Value(182)
+Enum(avr_mcu) String(atmega1284p) Value(182)
 
 EnumValue
-Enum(avr_mcu) String(at90usb1286) Value(183)
+Enum(avr_mcu) String(atmega128rfa1) Value(183)
 
 EnumValue
-Enum(avr_mcu) String(at90usb1287) Value(184)
+Enum(avr_mcu) String(at90can128) Value(184)
 
 EnumValue
-Enum(avr_mcu) String(avr6) Value(185)
+Enum(avr_mcu) String(at90usb1286) Value(185)
 
 EnumValue
-Enum(avr_mcu) String(atmega2560) Value(186)
+Enum(avr_mcu) String(at90usb1287) Value(186)
 
 EnumValue
-Enum(avr_mcu) String(atmega2561) Value(187)
+Enum(avr_mcu) String(avr6) Value(187)
 
 EnumValue
-Enum(avr_mcu) String(avrxmega2) Value(188)
+Enum(avr_mcu) String(atmega2560) Value(188)
 
 EnumValue
-Enum(avr_mcu) String(atxmega16a4) Value(189)
+Enum(avr_mcu) String(atmega2561) Value(189)
 
 EnumValue
-Enum(avr_mcu) String(atxmega16d4) Value(190)
+Enum(avr_mcu) String(avrxmega2) Value(190)
 
 EnumValue
-Enum(avr_mcu) String(atxmega32a4) Value(191)
+Enum(avr_mcu) String(atxmega16a4) Value(191)
 
 EnumValue
-Enum(avr_mcu) String(atxmega32d4) Value(192)
+Enum(avr_mcu) String(atxmega16d4) Value(192)
 
 EnumValue
-Enum(avr_mcu) String(atxmega32x1) Value(193)
+Enum(avr_mcu) String(atxmega32a4) Value(193)
 
 EnumValue
-Enum(avr_mcu) String(atmxt112sl) Value(194)
+Enum(avr_mcu) String(atxmega32d4) Value(194)
 
 EnumValue
-Enum(avr_mcu) String(atmxt224) Value(195)
+Enum(avr_mcu) String(atxmega32x1) Value(195)
 
 EnumValue
-Enum(avr_mcu) String(atmxt224e) Value(196)
+Enum(avr_mcu) String(atmxt112sl) Value(196)
 
 EnumValue
-Enum(avr_mcu) String(atmxt336s) Value(197)
+Enum(avr_mcu) String(atmxt224) Value(197)
 
 EnumValue
-Enum(avr_mcu) String(atxmega16a4u) Value(198)
+Enum(avr_mcu) String(atmxt224e) Value(198)
 
 EnumValue
-Enum(avr_mcu) String(atxmega16c4) Value(199)
+Enum(avr_mcu) String(atmxt336s) Value(199)
 
 EnumValue
-Enum(avr_mcu) String(atxmega32a4u) Value(200)
+Enum(avr_mcu) String(atxmega16a4u) Value(200)
 
 EnumValue
-Enum(avr_mcu) String(atxmega32c4) Value(201)
+Enum(avr_mcu) String(atxmega16c4) Value(201)
 
 EnumValue
-Enum(avr_mcu) String(atxmega32e5) Value(202)
+Enum(avr_mcu) String(atxmega32a4u) Value(202)
 
 EnumValue
-Enum(avr_mcu) String(avrxmega4) Value(203)
+Enum(avr_mcu) String(atxmega32c4) Value(203)
 
 EnumValue
-Enum(avr_mcu) String(atxmega64a3) Value(204)
+Enum(avr_mcu) String(atxmega32e5) Value(204)
 
 EnumValue
-Enum(avr_mcu) String(atxmega64d3) Value(205)
+Enum(avr_mcu) String(avrxmega4) Value(205)
 
 EnumValue
-Enum(avr_mcu) String(atxmega64a3u) Value(206)
+Enum(avr_mcu) String(atxmega64a3) Value(206)
 
 EnumValue
-Enum(avr_mcu) String(atxmega64a4u) Value(207)
+Enum(avr_mcu) String(atxmega64d3) Value(207)
 
 EnumValue
-Enum(avr_mcu) String(atxmega64b1) Value(208)
+Enum(avr_mcu) String(atxmega64a3u) Value(208)
 
 EnumValue
-Enum(avr_mcu) String(atxmega64b3) Value(209)
+Enum(avr_mcu) String(atxmega64a4u) Value(209)
 
 EnumValue
-Enum(avr_mcu) String(atxmega64c3) Value(210)
+Enum(avr_mcu) String(atxmega64b1) Value(210)
 
 EnumValue
-Enum(avr_mcu) String(atxmega64d4) Value(211)
+Enum(avr_mcu) String(atxmega64b3) Value(211)
 
 EnumValue
-Enum(avr_mcu) String(avrxmega5) Value(212)
+Enum(avr_mcu) String(atxmega64c3) Value(212)
 
 EnumValue
-Enum(avr_mcu) String(atxmega64a1) Value(213)
+Enum(avr_mcu) String(atxmega64d4) Value(213)
 
 EnumValue
-Enum(avr_mcu) String(atxmega64a1u) Value(214)
+Enum(avr_mcu) String(avrxmega5) Value(214)
 
 EnumValue
-Enum(avr_mcu) String(avrxmega6) Value(215)
+Enum(avr_mcu) String(atxmega64a1) Value(215)
 
 EnumValue
-Enum(avr_mcu) String(atxmega128a3) Value(216)
+Enum(avr_mcu) String(atxmega64a1u) Value(216)
 
 EnumValue
-Enum(avr_mcu) String(atxmega128d3) Value(217)
+Enum(avr_mcu) String(avrxmega6) Value(217)
 
 EnumValue
-Enum(avr_mcu) String(atxmega192a3) Value(218)
+Enum(avr_mcu) String(atxmega128a3) Value(218)
 
 EnumValue
-Enum(avr_mcu) String(atxmega192d3) Value(219)
+Enum(avr_mcu) String(atxmega128d3) Value(219)
 
 EnumValue
-Enum(avr_mcu) String(atxmega256a3) Value(220)
+Enum(avr_mcu) String(atxmega192a3) Value(220)
 
 EnumValue
-Enum(avr_mcu) String(atxmega256a3b) Value(221)
+Enum(avr_mcu) String(atxmega192d3) Value(221)
 
 EnumValue
-Enum(avr_mcu) String(atxmega256a3bu) Value(222)
+Enum(avr_mcu) String(atxmega256a3) Value(222)
 
 EnumValue
-Enum(avr_mcu) String(atxmega256d3) Value(223)
+Enum(avr_mcu) String(atxmega256a3b) Value(223)
 
 EnumValue
-Enum(avr_mcu) String(atxmega128a3u) Value(224)
+Enum(avr_mcu) String(atxmega256a3bu) Value(224)
 
 EnumValue
-Enum(avr_mcu) String(atxmega128b1) Value(225)
+Enum(avr_mcu) String(atxmega256d3) Value(225)
 
 EnumValue
-Enum(avr_mcu) String(atxmega128b3) Value(226)
+Enum(avr_mcu) String(atxmega128a3u) Value(226)
 
 EnumValue
-Enum(avr_mcu) String(atxmega128c3) Value(227)
+Enum(avr_mcu) String(atxmega128b1) Value(227)
 
 EnumValue
-Enum(avr_mcu) String(atxmega128d4) Value(228)
+Enum(avr_mcu) String(atxmega128b3) Value(228)
 
 EnumValue
-Enum(avr_mcu) String(atmxt540s) Value(229)
+Enum(avr_mcu) String(atxmega128c3) Value(229)
 
 EnumValue
-Enum(avr_mcu) String(atmxt540sreva) Value(230)
+Enum(avr_mcu) String(atxmega128d4) Value(230)
 
 EnumValue
-Enum(avr_mcu) String(atxmega192a3u) Value(231)
+Enum(avr_mcu) String(atmxt540s) Value(231)
 
 EnumValue
-Enum(avr_mcu) String(atxmega192c3) Value(232)
+Enum(avr_mcu) String(atmxt540sreva) Value(232)
 
 EnumValue
-Enum(avr_mcu) String(atxmega256a3u) Value(233)
+Enum(avr_mcu) String(atxmega192a3u) Value(233)
 
 EnumValue
-Enum(avr_mcu) String(atxmega256c3) Value(234)
+Enum(avr_mcu) String(atxmega192c3) Value(234)
 
 EnumValue
-Enum(avr_mcu) String(atxmega384c3) Value(235)
+Enum(avr_mcu) String(atxmega256a3u) Value(235)
 
 EnumValue
-Enum(avr_mcu) String(atxmega384d3) Value(236)
+Enum(avr_mcu) String(atxmega256c3) Value(236)
 
 EnumValue
-Enum(avr_mcu) String(avrxmega7) Value(237)
+Enum(avr_mcu) String(atxmega384c3) Value(237)
 
 EnumValue
-Enum(avr_mcu) String(atxmega128a1) Value(238)
+Enum(avr_mcu) String(atxmega384d3) Value(238)
 
 EnumValue
-Enum(avr_mcu) String(atxmega128a1u) Value(239)
+Enum(avr_mcu) String(avrxmega7) Value(239)
 
 EnumValue
-Enum(avr_mcu) String(atxmega128a4u) Value(240)
+Enum(avr_mcu) String(atxmega128a1) Value(240)
 
 EnumValue
-Enum(avr_mcu) String(avr1) Value(241)
+Enum(avr_mcu) String(atxmega128a1u) Value(241)
 
 EnumValue
-Enum(avr_mcu) String(at90s1200) Value(242)
+Enum(avr_mcu) String(atxmega128a4u) Value(242)
 
 EnumValue
-Enum(avr_mcu) String(attiny11) Value(243)
+Enum(avr_mcu) String(avr1) Value(243)
 
 EnumValue
-Enum(avr_mcu) String(attiny12) Value(244)
+Enum(avr_mcu) String(at90s1200) Value(244)
 
 EnumValue
-Enum(avr_mcu) String(attiny15) Value(245)
+Enum(avr_mcu) String(attiny11) Value(245)
 
 EnumValue
-Enum(avr_mcu) String(attiny28) Value(246)
+Enum(avr_mcu) String(attiny12) Value(246)
+
+EnumValue
+Enum(avr_mcu) String(attiny15) Value(247)
+
+EnumValue
+Enum(avr_mcu) String(attiny28) Value(248)
 
diff -rupN gcc-4.9.2/gcc/config/avr/t-multilib gcc-4.9.2-patched/gcc/config/avr/t-multilib
--- gcc-4.9.2/gcc/config/avr/t-multilib	2014-10-30 09:28:58.000000000 +0100
+++ gcc-4.9.2-patched/gcc/config/avr/t-multilib	2014-12-18 19:29:21.000000000 +0100
@@ -63,8 +63,10 @@ MULTILIB_MATCHES = \
 	mmcu?avr25=mmcu?attiny4313 \
 	mmcu?avr25=mmcu?attiny44 \
 	mmcu?avr25=mmcu?attiny44a \
+	mmcu?avr25=mmcu?attiny441 \
 	mmcu?avr25=mmcu?attiny84 \
 	mmcu?avr25=mmcu?attiny84a \
+	mmcu?avr25=mmcu?attiny841 \
 	mmcu?avr25=mmcu?attiny25 \
 	mmcu?avr25=mmcu?attiny45 \
 	mmcu?avr25=mmcu?attiny85 \
diff -rupN gcc-4.9.2/gcc/doc/avr-mmcu.texi gcc-4.9.2-patched/gcc/doc/avr-mmcu.texi
--- gcc-4.9.2/gcc/doc/avr-mmcu.texi	2014-03-03 17:04:47.000000000 +0100
+++ gcc-4.9.2-patched/gcc/doc/avr-mmcu.texi	2014-12-18 19:29:23.000000000 +0100
@@ -18,7 +18,7 @@
 
 @item avr25
 ``Classic'' devices with up to 8@tie{}KiB of program memory and with the @code{MOVW} instruction.
-@*@var{mcu}@tie{}= @code{ata5272}, @code{ata6289}, @code{attiny13}, @code{attiny13a}, @code{attiny2313}, @code{attiny2313a}, @code{attiny24}, @code{attiny24a}, @code{attiny25}, @code{attiny261}, @code{attiny261a}, @code{attiny43u}, @code{attiny4313}, @code{attiny44}, @code{attiny44a}, @code{attiny45}, @code{attiny461}, @code{attiny461a}, @code{attiny48}, @code{attiny84}, @code{attiny84a}, @code{attiny85}, @code{attiny861}, @code{attiny861a}, @code{attiny87}, @code{attiny88}, @code{at86rf401}.
+@*@var{mcu}@tie{}= @code{ata5272}, @code{ata6289}, @code{attiny13}, @code{attiny13a}, @code{attiny2313}, @code{attiny2313a}, @code{attiny24}, @code{attiny24a}, @code{attiny25}, @code{attiny261}, @code{attiny261a}, @code{attiny43u}, @code{attiny4313}, @code{attiny44}, @code{attiny44a}, @code{attiny441}, @code{attiny45}, @code{attiny461}, @code{attiny461a}, @code{attiny48}, @code{attiny84}, @code{attiny84a}, @code{attiny841}, @code{attiny85}, @code{attiny861}, @code{attiny861a}, @code{attiny87}, @code{attiny88}, @code{at86rf401}.
 
 @item avr3
 ``Classic'' devices with 16@tie{}KiB up to 64@tie{}KiB of  program memory.
